PWGCF/TwoParticleCorrelations/Tasks/pidDiHadron.cxx

```diff
@@ -55,17 +57,109 @@ using namespace o2::framework::expressions;
 // define the filtered collisions and tracks
 #define O2_DEFINE_CONFIGURABLE(NAME, TYPE, DEFAULT, HELP) Configurable<TYPE> NAME{#NAME, DEFAULT, HELP};
 
+template <typename T, typename P>
+auto readMatrix(Array2D<T> const& mat, P& array, int rowStart, int rowEnd, int colStart, int colEnd)
+{
+  for (auto i = rowStart; i < rowEnd; ++i) {
+    for (auto j = colStart; j < colEnd; ++j) {
+      array[i][j] = mat(i, j);
+    }
+  }
+
+  return;
+}
+
+static constexpr float LongArrayFloat[3][20] = {{1.1, 1.2, 1.3, -1.1, -1.2, -1.3, 1.1, 1.2, 1.3, -1.1, -1.2, -1.3, 1.1, 1.2, 1.3, -1.1, -1.2, -1.3, 1.1, 1.2}, {2.1, 2.2, 2.3, -2.1, -2.2, -2.3, 1.1, 1.2, 1.3, -1.1, -1.2, -1.3, 1.1, 1.2, 1.3, -1.1, -1.2, -1.3, 1.1, 1.2}, {3.1, 3.2, 3.3, -3.1, -3.2, -3.3, 1.1, 1.2, 1.3, -1.1, -1.2, -1.3, 1.1, 1.2, 1.3, -1.1, -1.2, -1.3, 1.1, 1.2}};
+static constexpr int LongArrayInt[3][20] = {{1, 1, 1, -1, -1, -1, 1, 1, 1, -1, -1, -1, 1, 1, 1, -1, -1, -1, 1, 1}, {2, 2, 2, -2, -2, -2, 1, 1, 1, -1, -1, -1, 1, 1, 1, -1, -1, -1, 1, 1}, {3, 3, 3, -3, -3, -3, 1, 1, 1, -1, -1, -1, 1, 1, 1, -1, -1, -1, 1, 1}};
+
 struct PidDiHadron {
+  o2::aod::ITSResponse itsResponse;
   Service<ccdb::BasicCCDBManager> ccdb;
 
-  O2_DEFINE_CONFIGURABLE(cfgCutVtxZ, float, 10.0f, "Accepted z-vertex range")
-  O2_DEFINE_CONFIGURABLE(cfgCutPtMin, float, 0.2f, "minimum accepted track pT")
-  O2_DEFINE_CONFIGURABLE(cfgCutPtMax, float, 10.0f, "maximum accepted track pT")
+  enum ResoParticles {
+    K0 = 0,
+    LAMBDA = 1,
+    PHI = 2,
+    NResoParticles = 3
+  };
+  enum ParticleCuts {
+    kCosPA = 0,
+    kMassMin,
+    kMassMax,
+    kPosTrackPt,
+    kNegTrackPt,
+    kDCAPosToPVMin,
+    kDCANegToPVMin,
+    kLifeTime,
+    kRadiusMin,
+    kRadiusMax,
+    kRapidity,
+    kNParticleCuts
+  };
+  enum ParticleSwitches {
+    kUseCosPA = 0,
+    kMassBins,
+    kDCABetDaug,
+    kUseProperLifetime,
+    kUseV0Radius,
+    kNParticleSwitches
+  };
+  enum Particles {
+    PIONS = 0,
+    KAONS,
+    PROTONS
```

Comment:
I still believe this is inconsistent a dangerous
Depending on the part of the code you look at this assignment is true or is false
I understand you try to use a single configurable to address identified or not particles and which of the identified particle is of interest but this clearly creates an inconsistency between parts of your code
I would suggest trying to address that situation in a different way